### DISCUSSION TITLE
Things working/not workig and maybe workarounds?

### DISCUSSION THREAD

**Discussion by hpfmn** [danvd/wlroots-eglstreams#5] [General]
Hi,

I think it is better to use this discussion feature here. As written before qutebrowser is not working for me. Firefox seems to be the only browser with native wayland support that works. Furthermore thunderbird is quite broken for me with wayland. The GUI just gets partly rendered.

**Comment by hpfmn** [comment]
workaround for qutebrowser - use cli option "-s qt.force_software_rendering chromium"

**Comment by danvd** [comment]
As for me, chromium/chrome has pretty decent support for wayland. It could be started up with:
`--enable-features=UseOzonePlatform --ozone-platform=wayland`

**Reply by hpfmn** [reply]
Yeah it kinda does work. But is really unreliable for me. I use a scaling factor of two on the second display and if I switch between the first and the second it is pixelated and hangs. :(

But if I don't switch it to the other display it does work.

**Reply by hpfmn** [reply]
If you have any idea how to debug this let me know.

**Reply by danvd** [reply]
Your setup is 2 monitors with different DPI? By switching what do you mean?

**Reply by hpfmn** [reply]
Yeah. I have two diffrent DPI (Scale Factor of 2) by switching I mean moving the window between the monitors. Form HIDPI to low-DPI and back again. This seems to cause chromium to hang.

**Comment by danvd** [comment]
I tried qutebrowser. It falls back to software rendering:
```
[228576:228609:0309/072522.427406:ERROR:gl_context_egl.cc(214)] eglCreateContext failed with error EGL_BAD_CONTEXT
[228576:228609:0309/072522.427422:ERROR:gpu_channel_manager.cc(692)] ContextResult::kFatalFailure: Failed to create shared context for virtualization.
```
 I'm not sure why this happens. (Will look into its source when I have time.)

**Reply by hpfmn** [reply]
It seems qt5-webengine should gain more wayland support in 5.15.4

**Comment by danvd** [comment]
Just tried ozone wayland platform, but with no luck:
```
~ > qutebrowser --qt-flag enable-features=UseOzonePlatform --qt-flag ozone-platform=wayland
20:22:17 INFO: Run :adblock-update to get adblock lists.
[263010:263010:0309/202217.854882:FATAL:platform_selection.cc(48)] Invalid ozone platform: wayland
```
Looks like qt's version of chromium is too outdated, so qutebrowser is stuck on xwayland.
The latter still uses software rendering. Hopefully nvidia's dma-buf support + https://gitlab.freedesktop.org/xorg/xserver/-/merge_requests/587
will come soon.

**Comment by hpfmn** [comment]
Also electron apps do work with the beta of electron12 - but they are pixelated on 2.0 scaling for some reason.

**Reply by hpfmn** [reply]
Just updated to the most recent electron-12 release and the cli-flags ` --enable-features=UseOzonePlatform --ozone-platform=wayland` do work here :)

**Reply by hpfmn** [reply]
But it has the same strange behaviour like chrome (see above)

**Comment by hpfmn** [comment]
I just noted that copy and paste from images does not really work well. Is that also nvidia related?

**Reply by danvd** [reply]
Could you provide some details about the source and destination of copying?

**Reply by hpfmn** [reply]
In Okular you can copy part of a PDF to as an image to the clipboard and paste that in another application. Joplin (with electron12 and wayland enabled) wouldn't let me paste at all and in other programs the image was corrupted. If I would start in in X11 mode I could paste to X11 Apps though. Maybe this functionality somehow needs these nvidia-related buffer stuff for taking screenshots?

**Reply by danvd** [reply]
I'll try to test it with amd gpu when i have some time to swap cards

**Comment by mesufesu** [comment]
Is low performance inevitable? Having just one tab open in chrome results in 35% sway CPU usage, up to 50%. Playing some 1080p youtube video I get occasional frame drops windowed, but if I go fullscreen it starts to drop constantly to the point of stuttering. I can't test in Plasma and Mutter because Chromium doesn't even launch in them, but my AMD workstation with ancient 2C/4T Broadwell-E and RX570 and 4k+1080p screens shows 3-4% sway CPU utilization with a hundreds of Chromium tabs and a video playing in Firefox simultaneously. My NVIDIA machine is 3800X with 2070S (2x1440p) so the performance gap is unmistakably huge. Can't test Chromium playback with AMD at the moment because of the bug in 5.11 kernel.

In the same vein, mpv: with sway I get about 220 fps (refresh rate is 240Hz), while in X11 I don't know the ceiling with the same opengl-es api. This is probably fine because I tested mpv in eglstreams mutter also, and while the fps is the same 220, the mpv CPU utilization was about 40% lower, however.
